rust_src/src/buffers.rs

```diff
@@ -367,22 +367,12 @@ impl LispBufferRef {
 impl LispBufferRef {
     #[inline]
     pub fn overlays_before(&self) -> Option<LispOverlayRef> {
-        let p = self.overlays_before;
-        if p == ptr::null_mut() {
-            None
-        } else {
-            Some(ExternalPtr::new(p))
-        }
+        unsafe { self.overlays_before.as_ref().map(|m| mem::transmute(m)) }
```

Comment:
`|m| mem::transmute(m))` can be replaced with just `mem::transmute`, since it's already a function of one variable.

(It would be great if we could get Clippy working, because it finds stuff like that automatically.)

Reply:
Same below

Reply:
I get this
```
370 |         unsafe { self.overlays_before.as_ref().map(mem::transmute) }
    |                                                ^^^ expected an `FnOnce<(&remacs_sys::Lisp_Overlay,)>` closure, found `unsafe extern "rust-intrinsic" fn(_) -> _ {std::intrinsics::transmute::<_, _>}`
```

Reply:
Oh, nevermind then :)

Reply:
Clippy also has the benefit of preventing many eyeball review errors.